### ISSUE TITLE
train llama-2-7b-hf loss is 0 by 4bit

### ISSUE THREAD

**Issue by moseshu** [lvwerra/trl#618]
this is my training code [https://github.com/moseshu/deeplearning/blob/main/llama-finetune/llama2_sft_qrlora.py](url)
why the loss is not stable. for several epochs ,it's 0?why?
training on A100 GPU 
```
{'loss': 0.684, 'learning_rate': 3e-05, 'epoch': 0.31}
{'loss': 0.642, 'learning_rate': 3e-05, 'epoch': 0.62}
{'loss': 1.1291, 'learning_rate': 3e-05, 'epoch': 0.93}
{'loss': 1.2843, 'learning_rate': 3e-05, 'epoch': 1.25}
{'loss': 31.4517, 'learning_rate': 3e-05, 'epoch': 1.56}
{'loss': 0.0, 'learning_rate': 3e-05, 'epoch': 1.87}
{'loss': 0.0, 'learning_rate': 3e-05, 'epoch': 2.19}
......
.....
```

**Comment by lvwerra** [comment]
Hi there! Can you try to use `bfloat16` instead of `float16` for `bnb_4bit_compute_dtype`? We had some issues previously with that. Also make sure that you are right padding.

**Comment by moseshu** [comment]
> Hi there! Can you try to use `bfloat16` instead of `float16` for `bnb_4bit_compute_dtype`? We had some issues previously with that. Also make sure that you are right padding.

Yes,I also tried bf16 ,the loss is still 0
